modules/core/src/main/scala/org/tessellation/infrastructure/rewards/Rewards.scala

```diff
@@ -67,28 +64,22 @@ object Rewards {
       def mintedDistribution(
         epochProgress: EpochProgress,
         facilitators: NonEmptySet[Id]
-      ): F[SortedSet[RewardTransaction]] = {
-        val amount = getAmountByEpoch(epochProgress, rewardsPerEpoch)
-
-        val programRewardsState = for {
-          stardustCollective <- stardust.distribute()
-          dtmRewards <- dtm.distribute()
-          softStakingRewards <- softStaking.distribute(epochProgress, facilitators)
-        } yield dtmRewards ++ softStakingRewards ++ stardustCollective
-
-        def eitherToF[A](either: Either[ArithmeticException, A]): F[A] = either.liftTo[F]
-
-        val allRewardsState = for {
-          programRewards <- programRewardsState.mapK[F](lift(eitherToF))
-          randomizer <- StateT.liftF(Random.scalaUtilRandomSeedLong(epochProgress.coerce))
-          regularRewards <- regular.distribute(randomizer, facilitators)
-        } yield programRewards ++ regularRewards
-
-        allRewardsState
-          .run(amount)
-          .flatTap(validateState(amount))
-          .map(toTransactions)
-      }
+      ): F[SortedSet[RewardTransaction]] =
+        Random.scalaUtilRandomSeedLong(epochProgress.coerce).flatMap { random =>
+          def eitherToF[A](either: Either[ArithmeticException, A]): F[A] = either.liftTo[F]
+
+          val allRewardsState = for {
+            programRewards <- programsDistributor.distribute().mapK[F](lift(eitherToF))
```

Comment:
is `eitherToF` used once only? won't `lift(_.liftTo[F])` work?

Reply:
`lift(_.liftTo[F])` won't compile. However, I see that `liftFunction(_.liftTo[F])` does compile. Fixing it.